api/src/cache.ts

```diff
@@ -61,6 +100,28 @@ export async function getSystemCache(key: string): Promise<Record<string, any>>
 	return await getCacheValue(systemCache, key);
 }
 
+export async function setSchemaCache(schema: SchemaOverview): Promise<void> {
+	const { localSchemaCache, sharedSchemaCache } = getCache();
+	const schemaHash = await getSimpleHash(JSON.stringify(schema));
+
+	await sharedSchemaCache.set('hash', schemaHash);
+
+	await localSchemaCache.set('schema', schema);
+	await localSchemaCache.set('hash', schemaHash);
+}
+
+export async function getSchemaCache(): Promise<SchemaOverview | undefined> {
+	const { localSchemaCache, sharedSchemaCache } = getCache();
+
+	const sharedSchemaHash = await sharedSchemaCache.get('hash');
+	if (!sharedSchemaHash) return;
```

Comment:
I think there's no issue, but want to confirm:
If `CACHE_ENABLED=false` but we already have schema cached, this condition will not met and execution will jump to :119 right?
I think the default behaviour is, if no cache configuration is set, schema cache will use memory.

Reply:
> If CACHE_ENABLED=false but we already have schema cached, this condition will not met and execution will jump to :119 right?

`CACHE_ENABLED` has no effect on the schema cache. Schema caching can be disabled via `CACHE_SCHEMA=false`. 

https://github.com/directus/directus/blob/a7398c28d72aa30dc58cb5d3eb4bf25bb5f7c6a7/api/src/utils/get-schema.ts#L31-L35

If `CACHE_SCHEMA=false` it will not fetch from cache at all, meaning this function will not be triggered. The schema will be fetched directly from the DB.

https://github.com/directus/directus/blob/a7398c28d72aa30dc58cb5d3eb4bf25bb5f7c6a7/api/src/utils/get-schema.ts#L51-L53


> if no cache configuration is set, schema cache will use memory.

Yes that is correct, the `CACHE_STORE` defaults to memory. 👍